db/main/migrate/20180329000000_create_github_installations.rb

```diff
@@ -0,0 +1,32 @@
+class CreateGithubInstallations < ActiveRecord::Migration
+  PSQL = ActiveRecord::Base.connection.select_value('SELECT version()')
+
+  def self.up
+    create_table :github_installations do |t|
+      t.belongs_to :owner, :polymorphic => true
+      t.string     :owner_type
+      t.integer    :owner_id
+      if /PostgreSQL 9.3/.match(PSQL)
+        t.json       :permissions
+      else
+        t.jsonb    :permissions
+      end
+      t.integer    :added_by
+      t.datetime   :deleted_on
+      t.datetime   :updated_on
+    end
+
+    def change
+      add_reference :github_installations, :added_by, foreign_key: { to_table: :users }
+      add_reference :github_installations, :deleted_by, foreign_key: { to_table: :users }
+      add_column :repositories, :active_on_org, :boolean
```

Comment:
@kerrizor should we use `active_on_org` or `locked`?
technically a repo could be active on org, but be unlocked for usage on .com?
the concept of `locked` is that all build events will be ignored while locked.

Reply:
First blush, I like `active_on_org` here because it described a state, while `locked` is a logical response to that state (and thus, should probably live in the code as a method.)

```
def locked?
  active_on_org
end
```

Reply:
@josh Just so I'm clear, can you explain what this column is actually for. 

Is it for distinguishing where Travis should run a build for it? On .org _or_ .com?

This is something we need for the Merge - to know which platform to run builds on.
Customers using our GH App will be _forced_ to move their repos over to .com, which will make them `:active_on_org: false`